Kickstarter-iOS/Features/MessageBanner/Controller/MessageBannerViewController.swift

```diff
@@ -19,7 +20,19 @@ public final class MessageBannerViewController: UIViewController, NibLoading {
   @IBOutlet fileprivate var messageLabel: UILabel!
 
   private var bannerType: MessageBannerType?
-  private var dismissible: Bool = true
+
+  public enum DismissType {
+    // Banner cannot be dismissed and will not be dismissed automatically.
+    case persist
+    // Banner can be dismissed by the user and will be dismissed automatically.
+    // Once the banner is dismissed, the delegate method `messageBannerViewDidHide` is called.
+    case bannerOnly
+    // Banner can be dismissed by the user and will be dismissed automatically.
+    // Once the banner is dismissed, the presenting view controller is popped.
+    case bannerAndParentVC
+  }
+
+  private var dismissType: DismissType = .bannerOnly
```

Comment:
I didn't add any new tests around dismissType - I don't think our code is structured to make it easy to capture when view controllers dismiss in tests (please correct me if I'm wrong about that!). Alternatively, I could move all the dismiss logic into the view controller, but that would be a bigger refactor than I was hoping for. Open to suggestions here!